File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l = 0
        r = 0
        count = 0
        max1 = float('-inf')
        while(r < len(nums)):
            if nums[r] == 0:
                k-=1
            if k < 0:
                if nums[l] == 0:
                    k+=1
                l+=1
            if k >= 0:
                max1 = max(max1,r-l+1)
            r+=1
        return max1
            

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int l = 0;
        int r = 0;
        int [] freq = new int[2];
        int res = 0;
        while(r < nums.length){
            freq[nums[r]]+=1;
            if(freq[0] > k){
                freq[nums[l]]-=1;
                l+=1;
            }
            if(freq[0] <= k){
                res = Math.max(res,r - l + 1);
            }
            r+=1;
        }
        return res;
    }
}